### PR TITLE
feat(protocols): add compound v3

### PR DESCRIPTION
**Pull request by 0xruhum**

### CODE REVIEW

File: src/yieldOptions/providers/protocols/compoundv3.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        // WETH
+        "0x4200000000000000000000000000000000000006": "0x46e6b214b524310239732D51387075E0e70970bf"
+    },
+};
+
+const CompAddress = {
+    1: "0xc00e94Cb662C3520282E6f5717214004A7f26888",
+    42161: "0x354A6dA3fcde098F8389cad84b0182725c6C91dE",
+    137: "0x8505b9d2254A7Ae468c0E9dd10Ccea3A837aef5c",
+    // TODO: couldn't find the COMP addresses for BASE
```
Comment (RedVeil) [inline src/yieldOptions/providers/protocols/compoundv3.ts:35]:
```suggestion
    8453: "0x9e1028F5F1D5eDE59748FFceE5532509976840E0"
```

File: src/yieldOptions/providers/protocols/compoundv3.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
+import { Address, parseAbi, PublicClient } from "viem";
+import { Clients, IProtocol } from "./index.js";
+import { Yield } from "src/yieldOptions/types.js";
+
+// Compound V2 is mainnet only. Thus we can simplify a lot of stuff
```
Comment (RedVeil) [inline src/yieldOptions/providers/protocols/compoundv3.ts:5]:
```suggestion
```

File: src/yieldOptions/providers/protocols/compoundv3.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                    apy: Number(supplyCompRewardApr),
+                }
+            ]
+
+        };
+        return result;
+    }
+
+    getAssets(chainId: number): Promise<Address[]> {
+        if (chainId !== 1) throw new Error("Compound V2 is only available on Ethereum mainnet");
```
Comment (RedVeil) [inline src/yieldOptions/providers/protocols/compoundv3.ts:93]: This is probably not accurate anymore

File: src/yieldOptions/providers/protocols/compoundv3.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    private getTrackingIndexScale(client: PublicClient, cToken: Address): Promise<bigint> {
+        return client.readContract({
+            address: cToken,
+            abi: CTOKEN_ABI,
+            functionName: "trackingIndexScale"
+        });
+    }
+
+    private async getCompPrice(): Promise<bigint> {
+        if (!this.clients[1]) {
```
Comment (RedVeil) [inline src/yieldOptions/providers/protocols/compoundv3.ts:167]: Can we let the user know somehow that they have to always add a mainnet client?

Comment (0xruhum) [reply src/yieldOptions/providers/protocols/compoundv3.ts]: I've moved the check to the constructor. That way they will get an error message early on.

**Pull request merged**